symbulate/tests/test_distributions.py

```diff
@@ -10,25 +10,705 @@
+
+
+class TestWeibull(unittest.TestCase):
+    def test_Weibull_to_Exponential(self):
+        '''X = RV(Weibull(scale=1, c=10))
+           sims = X.sim(Nsim)
+           cdf = stats.expon(scale=1/10).cdf
+           pval = stats.kstest(sims, cdf).pvalue
+           self.assertTrue(pval > .01)
+        '''
+        pass
+
+class TestRayleigh(unittest.TestCase):
+
+    def test_Rayleigh_Normal(self):
+        A, B = RV(Normal(mean=0, var=1) * Normal(mean=0, var=1))
+        sims = (A ** 2 + B ** 2).apply(sqrt).sim(Nsim)
+        cdf = stats.rayleigh.cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > .01)
+
+    def test_Rayleigh_to_Chi(self):
+        X = RV(Rayleigh())
+        sims = X.sim(Nsim)
+        cdf = stats.chi(df=2).cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > .01)
+
+
+class TestMultivariateNormal(unittest.TestCase):
+     
+    def test_MultivariateNormal_mean_cov_error(self):
+        self.assertRaises(Exception, 
+             lambda: MultivariateNormal(mean=[2], cov=[[2,2], [3,4]]))
+
+    def test_MultivariateNormal_cov_square_error(self):
+        self.assertRaises(Exception, 
+             lambda: MultivariateNormal(mean=[2,4], cov=[[2,4,5],[2,1]]))
+
+
+class TestBivariateNormal(unittest.TestCase):
+    
+    def test_BivariateNormal_error1(self):
+        self.assertRaises(Exception, lambda: 
+             BivariateNormal(mean1=3, mean2=4, sd1=-3, sd2=3, corr=0.9))
+
+    def test_BivariateNormal_error2(self):
+        self.assertRaises(Exception, lambda: 
+             BivariateNormal(mean1=3, mean2=4, sd1=3, sd2=3, corr=1.1))
+ 
+    def test_LinCom_BivNormal(self):
+        X, Y = RV(BivariateNormal(mean1=30, mean2=50, sd1=8, sd2=6, corr=-0.4))
+        Z = 4 * X - 2 * Y
+        Z_mean = 4 * 30 + (-2) * 50
+        Z_var = 16 * 64 + 4 * 36 + 2 * (4) * (-2) * (-0.4 * 8 * 6)
+        sims = Z.sim(Nsim)
+        cdf = stats.norm(loc=Z_mean, scale=sqrt(Z_var)).cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > .01)
+    
+
+    def test_BivNormal_condDistr_r(self):
+        X,Y = RV(BivariateNormal(mean1=20, mean2=10, sd1=3, sd2=5, corr=0.5))
+        sims = (Y | (abs(X - 21) < 0.1)).sim(Nsim)
+        cdf = stats.norm(loc=10 + 0.5 * 5 / 3 * (21 - 20), 
+                         scale=5 * sqrt(1 - (0.5) ** 2)).cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > 0.01)
+
+    #if approx 1 (0.9999), it fails?
+    def test_BivNormal_condDistr_r(self):
+        X,Y = RV(BivariateNormal(mean1=20, mean2=10, sd1=3, sd2=5, corr=0.9))
+        sims = (Y | (abs(X - 21) < 0.1)).sim(1000)
+        cdf = stats.norm(loc=10 + 0.9 * 5 / 3 * (21 - 20), 
+                         scale=5 * sqrt(1 - (0.9) ** 2)).cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > 0.01)
+
+    #if approx -1 (-0.99999), it fails
+    def test_BivNormal_condDistr_r(self):
```

Comment:
It doesn't make sense to repeat this test 3 times, which is pretty much exactly the same, except for the value of the correlation. You should write a for loop over correlation values, i.e.,

```
for corr in [0.5, 0.9, -0.9]:
    ...
```

and for each one, calculate the p-value and assert that the p-value is > 0.01.